code/datums/martial/plasma_fist.dm

```diff
@@ -52,9 +61,50 @@
 	D.visible_message("<span class='danger'>[A] hits [D] with THE PLASMA FIST TECHNIQUE!</span>", \
 					"<span class='userdanger'>You're suddenly hit with THE PLASMA FIST TECHNIQUE by [A]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", null, A)
 	to_chat(A, "<span class='danger'>You hit [D] with THE PLASMA FIST TECHNIQUE!</span>")
+	var/turf/Dturf = get_turf(D)
 	D.gib()
 	log_combat(A, D, "gibbed (Plasma Fist)")
-	return
+	if(nobomb)
+		return
+	if(!isliving(target) || !target.mind)
+		to_chat(A, "<span class='warning'>You cannot power up your </span><span class='notice'>Apotheosis</span><span class='warning'> off of this!</span>")
+		return
+	else if(plasma_power >= plasma_cap)
+		to_chat(A, "<span class='warning'>You cannot power up your </span><span class='notice'>Apotheosis</span><span class='warning'> any more!</span>")
+		new /obj/effect/temp_visual/plasma_soul(Dturf)//doesn't beam to you, so it just hangs around and poofs.
+	else
+		plasma_power++
+		to_chat(A, "<span class='nicegreen'>Power increasing! Your </span><span class='notice'>Apotheosis</span><span class='nicegreen'> is now at power level [plasma_power]!</span>")
+		new /obj/effect/temp_visual/plasma_soul(Dturf, A)
+		var/oldcolor = A.color
+		A.color = "#9C00FF"
+		flash_color(A, flash_color = "#9C00FF", flash_time = 3 SECONDS)
+		animate(A, color = oldcolor, time = 3 SECONDS)
+	
+
+/datum/martial_art/plasma_fist/proc/Apotheosis(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	A.say("APOTHEOSIS!!", forced="plasma fist")
+	A.set_species(/datum/species/plasmaman)
+	A.dna.species.species_traits += TRAIT_BOMBIMMUNE //hey you get something out of it
+	A.unequip_everything()
+	A.underwear = "Nude"
+	A.undershirt = "Nude"
+	A.socks = "Nude"
+	A.update_body()
+	var/turf/boomspot = get_turf(A)
+
+	A.apply_damage(rand(50,70), BRUTE)
+	A.Stun(6 SECONDS) //stops them from comboing apotheosis twice (lol)
+	log_combat(A, A, "triggered final plasma explosion with size [plasma_power], [plasma_power*2], [plasma_power*4] (Plasma Fist)")
+	addtimer(CALLBACK(src,.proc/Apotheosis_end, A), 6 SECONDS)
+	playsound(boomspot, 'sound/weapons/punch1.ogg', 50, TRUE, -1)
+	explosion(boomspot,plasma_power,plasma_power*2,plasma_power*4,ignorecap = TRUE)
+
+/datum/martial_art/plasma_fist/proc/Apotheosis_end(mob/living/carbon/human/dying)
+	if(dying.stat == DEAD)
+		return
+	to_chat(dying, "<span class='userdanger'>Your body gives in.</span>")
+	dying.death()
```

Comment:
Can you get cloned and do it again? That would probably be bad.

Reply:
not only can you get cloned and do it again, but you'll be bomb proof if you're revived

Reply:
:smirk: